backend/fms_core/template_importer/importers/extraction.py

```diff
@@ -27,6 +27,11 @@ def initialize_data_for_template(self):
     def import_template_inner(self):
         sheet = self.sheets['ExtractionTemplate']
 
+        # Add the template to the process
+        if self.imported_file is not None:
+            self.preloaded_data['process'].imported_template_id = self.imported_file.id
+            self.preloaded_data['process'].save()
```

Comment:
DB stuff in services is not a religion ... i do not think it deserves a service since there is really nothing here to reuse. Service are more to buffer the internal core from input from templates and to gather commonly used functionality for complex objects.